flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+        :param Text memory_request:
+        :param Text storage_limit:
+        :param Text cpu_limit:
+        :param Text gpu_limit:
+        :param Text memory_limit:
+        :param dict[Text, Text] environment:
+        :rtype: flytekit.models.task.Container
+        """
+
+        storage_limit = storage_limit or storage_request
+        cpu_limit = cpu_limit or cpu_request
+        gpu_limit = gpu_limit or gpu_request
+        memory_limit = memory_limit or memory_request
+
+        requests = []
+        if storage_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.STORAGE,
+                    storage_request
+                )
+            )
+        if cpu_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.CPU,
+                    cpu_request
+                )
+            )
+        if gpu_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.GPU,
+                    gpu_request
+                )
+            )
+        if memory_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.MEMORY,
+                    memory_request
+                )
+            )
+
+        limits = []
+        if storage_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.STORAGE,
+                    storage_limit
+                )
+            )
+        if cpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.CPU,
+                    cpu_limit
+                )
+            )
+        if gpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.GPU,
+                    gpu_limit
+                )
+            )
+        if memory_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.MEMORY,
+                    memory_limit
+                )
+            )
+
+        return  _sdk_runnable.SdkRunnableContainer(
+            command=[],
+            args=[],
+            resources=_task_models.Resources(limits=limits, requests=requests),
+            env=environment,
+            config={}
+        )
+
+
+
+
+def spark_notebook(
+        cache_version='',
+        retries=0,
+        deprecated='',
+        cache=False,
+        timeout=None,
+        notebook_path=None,
+        inputs=None,
+        outputs=None,
+        environment=None,
+):
+    """
```

Comment:
lets add what is to be required and how to use it

For example, mark cell as conf, have a variable called spark_conf